File: src/main/java/no/nav/syfo/sts/StsToken.java
package no.nav.syfo.sts;

import lombok.Data;
import lombok.experimental.Accessors;

import java.time.LocalDateTime;

@Data
@Accessors(fluent = true)
public class StsToken {
    public String access_token;
    public String token_type;
    public int expires_in;

    LocalDateTime expirationTime;

    public static void setExpirationTime(StsToken token) {
        if (token == null) {
            return;
        }
        token.expirationTime = LocalDateTime.now().plusSeconds(token.expires_in - 10L);
    }

    public static boolean shouldRenew(StsToken token) {
        if (token == null) {
            return true;
        }
        return isExpired(token);
    }

    private static boolean isExpired(StsToken token) {
        return token.expirationTime.isBefore(LocalDateTime.now());
    }
}

File: src/main/kotlin/no/nav/syfo/sts/StsToken.kt
package no.nav.syfo.sts

import java.time.LocalDateTime

data class StsToken(
        val access_token: String,
        val token_type: String,
        val expires_in: Int
) {
    // Expire 10 seconds before token expiration

    val expirationTime: LocalDateTime = LocalDateTime.now().plusSeconds(expires_in - 10L)

    companion object {
        fun shouldRenew(token: StsToken?): Boolean {
            if (token == null) {
                return true
            }

            return isExpired(token)
        }

        private fun isExpired(token: StsToken): Boolean {
            return token.expirationTime.isBefore(LocalDateTime.now())
        }
    }
}